src/lib/core/WeaveConfig.h

```diff
@@ -1403,6 +1404,55 @@
 #define WEAVE_CONFIG_DEBUG_CERT_VALIDATION                  1
 #endif // WEAVE_CONFIG_DEBUG_CERT_VALIDATION
 
+/**
+ *  @def WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID
+ *
+ *  @brief
+ *    Default curve to be used to generate Weave operational device certificate.
+ *
+ */
+#ifndef WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID
+#if WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP256R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_prime256v1)
+#elif WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP224R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_secp224r1)
+#elif WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP192R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_prime192v1)
+#elif WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP160R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_secp160r1)
+#else
+#error "Specify valid supported Elliptic Curve to be used to generate Weave operation device certificate."
+#endif
+#endif // WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID
+
+/**
+ *  @def WEAVE_CONFIG_DEFAULT_CERT_VALID_DATE_NOT_BEFORE
+ *
+ *  @brief
+ *    This is the default valid date to be encoded in the
+ *    Weave operational device certificate. Any date before
+ *    that date the certificate is considered invalid.
+ *    Currently encoded value corresponds to July 15, 2019.
+ *
+ */
+#ifndef WEAVE_CONFIG_DEFAULT_CERT_VALID_DATE_NOT_BEFORE
+#define WEAVE_CONFIG_DEFAULT_CERT_VALID_DATE_NOT_BEFORE     0x1C64
```

Comment:
Is there an encoding macro that can be used to make this more scrutable / human-readable? If you asked me to update this date to January 7, 2020, I'd have no idea where to begin.

Reply:
Comment should note that this is a "packed" certificate date, and reference the PackCertTime() and PackedCertTimeToDate() functions.  It can also give the mathematical expression, which is

```
(((year - 2000) * 12 + (mon - 1)) * 31 + (day - 1))
```

Reply:
I also suggest switching to January 1st, just so that people aren't left wondering what's special about July 15th.

Reply:
Updated description.